container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/Node.java

```diff
@@ -21,13 +21,20 @@
     private final AtomicBoolean statusIsKnown = new AtomicBoolean(false);
     private final AtomicBoolean working = new AtomicBoolean(true);
     private final AtomicLong activeDocuments = new AtomicLong(0);
+    private final AtomicBoolean pendingPing = new AtomicBoolean();
 
     public Node(int key, String hostname, int group) {
         this.key = key;
         this.hostname = hostname;
         this.group = group;
     }
 
+    /** Only send ping if this method return true. If not the is a ping outstanding. */
+    public boolean sendPing() { return ! pendingPing.getAndSet(true); }
+
+    /** Need to be called when a pong is called to allow next ping to go through. */
+    public void receivePing() { pendingPing.set(false); }
```

Comment:
somewhat misleading name

Reply:
Take another look, Rewritten to use sequence numbers instead. Simpler and safer.